packages/orchestrator-ui-components/src/components/WFOWorkflowSteps/WFOStepList/WFOStepList.tsx

```diff
@@ -50,32 +62,36 @@ export const WFOStepList = React.forwardRef(
 
         return (
             <>
-                {steps.map((step, index) => {
+                {stepListItems.map((stepListItem, index) => {
                     let previousState = {};
                     let delta = {};
 
                     if (index > 0) {
                         // Todo: decided if we want this here. A refactor would be nice, or move to global state.
                         //  V1 has a rather big case statement with business logic here
-                        if (step.status === StepStatus.SUCCESS) {
-                            // Prepare a delta with the last successful step
+                        if (stepListItem.step.status === StepStatus.SUCCESS) {
+                            // Prepare a delta with the last successful stepListItem
                             let previousIndex = index - 1;
                             while (
                                 previousIndex > 0 &&
-                                (steps[previousIndex].status ===
+                                (stepListItems[previousIndex].step.status ===
                                     StepStatus.FAILED ||
-                                    steps[previousIndex].status ===
+                                    stepListItems[previousIndex].step.status ===
```

Comment:
Maybe assign stepListItems[previousIndex].step.status to a variable? instead of having it twice?

In general I stop reading code as soon as it involves array indices ;) This can almost always be abstracte away with higher-level libs like lodash (or whatever is the flavour of the day) to avoid of-by-one errors.

Reply:
Extracted a previousStep const.

About the array and indices: normally I would agree, but this code is temporary copied from our V1 app. It is waiting to be replaced by a new functionality in the backend to calculate the diff.